### ISSUE TITLE
Problem with adding actions to simpy.gram 

### ISSUE THREAD

**Issue by lysnikolaou** [gvanrossum/pegen#59]
Hello,

I am currently trying to add actions for the `pass` statement into `simpy.gram`. That was proven a little bit more difficult than I expected due to the following problem:

A simple statement can contain more than one statements, thus its return type should be `asdl_seq *`, which is propagated to the `statement` rule, whose return type also becomes `asdl_seq *`.
My question is, how can the `statements` rule handle such a return  type from `statement`? Going through the generated code, it seems like the loop for `statements` expects `stmt_ty` as the return type of `statement`.

The relevant grammar rules are these:
```
start[mod_ty]: a=[statements] ENDMARKER { Module(a, NULL, p->arena) }
statements[asdl_seq*]: a=statement+ { a }
statement[asdl_seq*]:  compound_stmt | simple_stmt
simple_stmt[asdl_seq*]: a=small_stmt b=further_small_stmt* [';'] NEWLINE { seq_insert_in_front(p, b, a) }
further_small_stmt[stmt_ty]: ';' a=small_stmt { a }
small_stmt[stmt_ty]: ( return_stmt | import_stmt | pass_stmt | raise_stmt | yield_stmt | assert_stmt | del_stmt | global_stmt | nonlocal_stmt
                     | assignment | expressions
                     )
pass_stmt[stmt_ty]: a='pass' { _Py_Pass(EXTRA(a, a)) }
```

Surely, the action for `statements` is wrong and should be something else, but I can't really figure out what.

Side Note: `seq_insert_in_front` is a function I wrote in `pegen.c`, which accepts the `asdl_seq *` generated by the loop in `simple_stmt` and prepends the first `small_stmt`.

**Comment by gvanrossum** [comment]
I think that we need a new helper function, one that flattens an array of arrays. Its argument is an `asdl_seq*` whose items are themselves all `asdl_seq*`, and the return value is an `asdl_seq*` containing the concatenated items of all those inner `asdl_seq*`s.

PS. I'd prefer it if `seq_insert_in_front` took its arguments in the order that they end up in the return sequence, i.e. `seq_insert_in_front(p, a, b)`.

**Comment by gvanrossum** [comment]
(Also, thanks for diving in this deep. This is exactly what I was hoping for!)

**Comment by lysnikolaou** [comment]
This can be probably closed, now that #60 is merged.

**Comment by gvanrossum** [comment]
Yup!
